src/main/kotlin/org/radarbase/push/integration/garmin/auth/GarminAuthValidator.kt

```diff
@@ -76,15 +76,21 @@ class GarminAuthValidator(
         }
     }
 
-    private fun checkIsAuthorised(userId: String, accessToken: String): Boolean {
+    private fun checkIsAuthorised(userId: String, accessToken: String, retry: Boolean = true):
+        Boolean {
         val user = try {
             userRepository.findByExternalId(userId)
         } catch (exc: NoSuchElementException) {
-            logger.warn(
-                "no_user: The user {} could not be found in the " +
-                    "user repository.", userId
-            )
-            return false
+            return if (retry) {
```

Comment:
Not sure if this path is liable for effectively a DoS attack, but I would also put a timeout on the retry. See e.g. https://github.com/RADAR-base/radar-jersey/blob/master/radar-jersey/src/main/kotlin/org/radarbase/jersey/util/CachedMap.kt
for an implementation.

Reply:
Ok added a timeout now.

Some other measures in place in the nginx reverse proxy -
- Rate limiting for garmin location
- IP address whitelisting for garmin location, restricting the IPs that can access the resource based on IPs provided by Garmin API support team.